WalletWasabi/BitcoinCore/P2pNode.cs

```diff
@@ -160,30 +162,45 @@ public void Disconnect()
 					}
 				}
 
-				try
-				{
-					node.Disconnect();
-				}
-				catch (Exception ex)
-				{
-					Logger.LogDebug(ex);
-				}
-				finally
+				await DisconnectAsync(node, cancel).ConfigureAwait(false);
+
+				Logger.LogInfo("P2p Bitcoin node is disconnected.");
+				return true;
+			}
+			catch (Exception ex)
+			{
+				Logger.LogError($"P2p Bitcoin node failed to disconnect. '{ex.ToUserFriendlyString()}'");
+				return false;
+			}
+			finally
+			{
+				Node = null;
```

Comment:
This comes from Visual Basic and from some early documentation writer misunderstanding how c# works. Setting fields to null makes no sense no matter what random ex-vb dinosaurs write about.

Reply:
Refactor P2pNode should be another PR. I don't like the current implementation either but that is not a goal here.